File: src/main/java/plazma/kernel/lib/data/xml/XmlParserContext.java
/*
 * Copyright (C) 2012-2023 Oleh Hapon ohapon@users.sourceforge.net
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307, USA.
 * 
 * Oleh Hapon
 * Kyiv, UKRAINE
 * ohapon@users.sourceforge.net
 */

package plazma.kernel.lib.data.xml;

import java.util.Map;

import plazma.kernel.lib.data.node.Node;
import plazma.kernel.lib.text.ParserContext;

public class XmlParserContext extends ParserContext {

    public int currEvent = 0;
    public int nodeEvent = 0;

    public String currTag;
    public String currAttribute;
    public Node currNode;
    
    public Map<Integer, String> map;

    public XmlParserContext() {
        super();
    }

    public String toString() {
        return "XmlParserContext[]";
    }

    public void destroy() {
        if (map != null) {
            map.clear();
            map = null;
        }
    }

}

File: src/main/cpp/plazma/lib/data/xml/XmlParserContext.cpp
#include <string>

#include "XmlParserContext.h"

namespace xml {

    XmlParserContext::XmlParserContext() {
        currEvent = 0;
        nodeEvent = 0;

        currTag = nullptr;
        currAttribute = nullptr;

        currNode = nullptr;
    }

    XmlParserContext::~XmlParserContext() {
        if (currTag != nullptr) {
            delete currTag;
        }
        if (currAttribute != nullptr) {
            delete currAttribute;
        }
        //cout << "Destroy XmlParserContext: " << endl;
        //destroy();
    }

    std::string XmlParserContext::toString() {
        return "XmlParserContext[]";
    }

}
